examples/autoqasm/5_Pulse_programming_and_dynamical_decoupling.ipynb

```diff
@@ -0,0 +1,263 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "8ebd3454",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@aq.main\n",
+    "def idle(qubit: int, idle_duration: float):\n",
+    "    control_frame = device.frames[f\"q{qubit}_rf_frame\"]\n",
+    "    pulse.delay(control_frame, idle_duration)\n",
+    "    pulse.capture_v0(control_frame)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "eeee0f33",
+   "metadata": {},
+   "source": [
+    "Below prints out the OpenQASM script of the program. The pulse instructions, delay and capture, are in a `cal` block, which stand for \"calibration\". A `cal` block defines the scope for low level hardware instructions. Today, the hardware instructions are pulse instructions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "3e76fdca",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "OPENQASM 3.0;\n",
+      "defcalgrammar \"openpulse\";\n",
+      "cal {\n",
+      "    delay[10.0us] q0_rf_frame;\n",
+      "    capture_v0(q0_rf_frame);\n",
+      "}\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(idle(0, 10e-6).to_ir())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2b55c594",
+   "metadata": {},
+   "source": [
+    "## Error supression with dynamical decoupling sequences"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "05d92b91",
+   "metadata": {},
+   "source": [
+    "In the `idle` program above, we intentionally add a delay instruction to the program. In more practical programs, qubit idling is inevitable. It often happens when a gate or pulse instruction applies on multiple qubits, but some of the qubit are still under other instructions. The rest of the qubits are idling and waiting for all qubits to be ready. During this idling time, the qubits are subjected to decoherence, resulting in low program fidelity. Dynamical decoupling is a technique to suppress such errors. It includes alternating pairs of X and Y \"$\\pi$ pulse\", a rotation around the Bloch sphere axis with an angle $\\pi$, applying on the qubits during the idling time. These pulses together are equivalent to an identity gate, but alternating \"$\\pi$ pulses\" make some of the decoherence effectively cancel each other out. To learn more about dynamical decoupling, [this blog post](https://aws.amazon.com/blogs/quantum-computing/suppressing-errors-with-dynamical-decoupling-using-pulse-control-on-amazon-braket/) has a detailed explanations and visualization of dynamical decoupling sequences."
```

Comment:
Can we also start with a short paragraph explaining which features of AutoQASM we'll cover? Because in the end this notebook isn't about DD. We should probably move the mention of the other blog post to the start as well, and explain how this notebook relates to that post